main.hpp

```diff
@@ -47,82 +60,149 @@ void printCsv(vector<IntersectedCell> &well_blocks) {
-             "path to model grid file (e.g. *.GRID)")
-            ("heel,h", po::value<vector<double>>()->multitoken(),
-             "Heel coordinates (x y z)")
-            ("toe,t", po::value<vector<double>>()->multitoken(),
-             "Toe coordinates (x y z)")
-            ("radius,r", po::value<double>(),
-             "wellbore radius")
+    ("help", "print help message")
+    ("grid,g", po::value<string>(),
+    "path to model grid file (e.g. *.GRID)")
+    ("heel,h", po::value<vector<double>>()->multitoken(),
+    "Heel coordinates (x y z)")
+    ("toe,t", po::value<vector<double>>()->multitoken(),
+    "Toe coordinates (x y z)")
+    ("radius,r", po::value<double>(),
+    "wellbore radius")
             ("compdat,c", po::value<int>()->implicit_value(0),
              "print in compdat format instead of CSV")
-            ("well-name,w", po::value<string>(),
-             "well name to be used when writing compdat")
-            ;
-	
+    ("well-name,w", po::value<string>(),
+    "well name to be used when writing compdat");
+
+    // Positional arguments <- OV
+    po::positional_options_description p;
+    p.add("grid", 1);
+
     // Process arguments to variable map
     po::variables_map vm;
-    
-    // Parse the input arguments and store the values 
-    po::store(po::parse_command_line(argc, argv, desc, po::command_line_style::unix_style ^ po::command_line_style::allow_short), vm);
-    
+
+    // Parse the input arguments and store the values
+    po::store(po::parse_command_line(argc, argv, desc,
+                                   po::command_line_style::unix_style ^
+                                   po::command_line_style::allow_short),
+                                   vm);
+
     // ??
     po::notify(vm);
 
-    // If called with --help or -h flag:
-    if (vm.count("help")) { // Print help if --help present or input file/output dir not present
-        cout << "Usage: ./WellIndexCalculator --grid gridpath --heel x1 y1 z1 --toe x2 y2 z2 --radius r [options]" << endl;
-        cout << desc << endl;
+    // Print help if --help/-h present or input file/output dir not present
+    string usage_msg = "Usage: ./WellIndexCalculator "
+                     "--grid gridpath "
+                     "--heel x1 y1 z1 "
+                     "--toe x2 y2 z2 "
+                     "--radius r [options]";
+
+    if (vm.count("help")) {
+        cout << usage_msg  << endl;
+        cout << desc       << endl;
         exit(EXIT_SUCCESS);
     }
 
-    assert(vm.count("grid"));
-    assert(vm.count("heel"));
-    assert(vm.count("toe"));
-    assert(vm.count("radius"));
-    if (vm.count("compdat"))
-        assert(vm.count("well-name"));
-    assert(vm["heel"].as<vector<double>>().size() == 3);
-    assert(vm["toe"].as<vector<double>>().size() == 3);
-    assert(boost::filesystem::exists(vm["grid"].as<string>()));
-    assert(vm["radius"].as<double>() > 0);
+    // Check input parameters are well-defined
```

Comment:
I'm really not a fan of this way of checking the input. This is exactly what `assert` is for. If you want to print a custom message, this should work: `assert(vm.count("grid") && "grid parameter is missing..." + usage_msg);`. The output will look like the following:
```
wicalc: /home/einar/Documents/GitHub/PCG/FieldOpt/FieldOpt/FieldOpt-WellIndexCalculator/main.hpp:144: boost::program_options::variables_map createVariablesMap(int, const char**): Assertion `vm.count("grid") && "Grid is missing"' failed.
[1]    5100 abort (core dumped)  ./wicalc
```
The message may be a bit messy, but it has the advantage of saying exactly where in the code the error occured. 

Alternatively, a way I actually like more than both is to create a custom assert macro that lets us specify the output format, while still being as easy to use as the standard `assert` macro. An example of how to implement one is given in [this answer](http://stackoverflow.com/a/37264642).

Also, indentation is a bit off. Use auto indent.